build/commands/lib/affectedTests.js

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+const { promisify } = require('util')
+const { readFile, writeFile } = require('fs/promises')
+const exec = promisify(require('child_process').execFile)
+const path = require('path')
+const config = require('./config')
+const { unlink } = require('fs-extra')
+const { randomUUID } = require('crypto')
+const { getApplicableFilters, getTestsToRun } = require('./testUtils')
+const { tmpdir } = require('os')
+
+const getTestTargets = (outDir, filters = ['//*']) => {
+  const { env, shell } = config.defaultOptions
+  return exec(
+    'gn',
+    ['ls', outDir, '--type=executable', '--testonly=true', ...filters],
+    { env, shell },
+  ).then((x) => x.stdout.trim().split('\n'))
+}
+
+const asGnTarget = (file) => {
+  const gnAbsolutePath = file.startsWith('//') ? file : '//brave/' + file
+
+  return gnAbsolutePath.replace('brave/chromium_src/', '')
+}
+
+async function getModifiedFiles(target = 'HEAD~', base = null) {
+  const args = ['diff', '--name-only', target, base].filter((x) => x)
+  const maxBuffer = 1024 * 1024 * 5
+  return exec('git', args, { maxBuffer }).then((x) =>
+    x.stdout
+      .trim()
+      .split('\n')
+      .filter((x) => x),
+  )
+}
+
+async function getReferenceCommit() {
```

Comment:
Probaby would have been better to use python for this stuff as there are already a lot of existing helpers for this kind of stuff in //third_party/depot_tools, but also most PRs are multiple commits. It should compare to whatever the base branch is for the current branch.

Reply:
Actually looks like it defaults to origin/master which is also problematic because your current branch might not be rebased on origin/master

Reply:
The commit ref is supplied via `--since_commit <commit/branch/tag>` I think this flag is more descriptive than just `--base`. `getReferenceCommit` is a fallback if no commit-ref is provided

Reply:
I think it's actually a bit confusing using `commit` in the name, but the less subjective reason to change it is consistency with other commands